src/customElements/components/markdown-editor/menu.ts

```diff
@@ -0,0 +1,75 @@
+import { Plugin } from 'prosemirror-state';
+import { EditorView } from 'prosemirror-view';
+import { MdMenuItemType, getMenuItemByType } from './menuItems';
+
+export type MenuItem = {
+    command: Function;
+    dom: HTMLButtonElement;
+    type: any;
+};
+
+export class MenuView {
+    editorView: EditorView;
+    items: MenuItem[];
+    dom: Element;
+
+    constructor(items: MenuItem[], editorView: EditorView, menuContainer: Element) {
+        this.items = items;
+        this.editorView = editorView;
+
+        this.dom = menuContainer;
+        items.forEach(({ dom }) => this.dom.appendChild(dom));
+        this.update();
+
+        this.dom.addEventListener('mousedown', (e: MouseEvent) => {
+            e.preventDefault();
+            editorView.focus();
+            items.forEach(({ command, dom }) => {
+                if (dom.contains(e.target as any))
+                    command(editorView.state, editorView.dispatch, editorView);
+            });
+        });
```

Comment:
this should be executed on connected callback.
event handling should be own method and unsubscribed on disconnected callback

Reply:
Doesn't it need to be a custom element to use connected callbacks?

Reply:
my bad